### PR TITLE
Review error messages in JSON

### PR DESCRIPTION
**Pull request by krwq**
Contributes to: https://github.com/dotnet/runtime/issues/63686

closes last bullet "Ensure exception messages are still relevant and up-to-date"

### CODE REVIEW

**Review by eiriktsarpalis — APPROVED** [review]
Thanks

**Comment by krwq** [comment]
changed the message per offline conversation

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Text.Json/src/Resources/Strings.resx [hunk 1]
```diff
@@ -576,7 +576,7 @@
     <value>An item with the same key has already been added. Key: {0}</value>
   </data>
   <data name="SerializerContextOptionsImmutable" xml:space="preserve">
-    <value>A 'JsonSerializerOptions' instance associated with a 'JsonSerializerContext' instance cannot be mutated once the context has been instantiated.</value>
+    <value>JsonSerializerOptions instances cannot be modified once encapsulated by a JsonSerializerContext. This can happen either when calling 'JsonSerializerOptions.AddContext' or when passing the options instance to a JsonSerializerContext constructor.</value>
```
[the remark this answers is not in the document]
Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/src/Resources/Strings.resx]: That's not actually accurate. It should probably read "Such encapsulation" instead.

File: src/libraries/System.Text.Json/src/Resources/Strings.resx [hunk 2]
```diff
@@ -576,7 +576,7 @@
     <value>An item with the same key has already been added. Key: {0}</value>
   </data>
   <data name="SerializerContextOptionsImmutable" xml:space="preserve">
-    <value>A 'JsonSerializerOptions' instance associated with a 'JsonSerializerContext' instance cannot be mutated once the context has been instantiated.</value>
+    <value>JsonSerializerOptions instances cannot be modified once encapsulated by a JsonSerializerContext. Such modification can happen either when calling 'JsonSerializerOptions.AddContext' or when passing the options instance to a JsonSerializerContext constructor.</value>
```
Comment (eiriktsarpalis) [inline src/libraries/System.Text.Json/src/Resources/Strings.resx:579]:
```suggestion
    <value>JsonSerializerOptions instances cannot be modified once encapsulated by a JsonSerializerContext. Such encapsulation can happen either when calling 'JsonSerializerOptions.AddContext' or when passing the options instance to a JsonSerializerContext constructor.</value>
```

File: src/libraries/System.Text.Json/src/Resources/Strings.resx (hunk 1 above)
[the remark this answers is not in the document]
Comment (stephentoub) [reply src/libraries/System.Text.Json/src/Resources/Strings.resx]: Changed. Proves the actual issue I was trying to address that "this" was ambiguous :-)

**Pull request merged**